src/network/networkd-manager-bus.c

```diff
@@ -3,17 +3,59 @@
 #include "alloc-util.h"
 #include "bus-util.h"
 #include "networkd-link-bus.h"
+#include "networkd-link.h"
 #include "networkd-manager-bus.h"
 #include "networkd-manager.h"
+#include "path-util.h"
 #include "strv.h"
 
+static int method_list_links(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        Manager *manager = userdata;
+        Iterator i;
+        Link *link;
+        int r;
+
+        r = sd_bus_message_new_method_return(message, &reply);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_open_container(reply, 'a', "(iso)");
+        if (r < 0)
+                return r;
+
+        HASHMAP_FOREACH(link, manager->links, i) {
+                _cleanup_free_ char *path = NULL;
+
+                path = link_bus_path(link);
+                if (!path)
+                        return -ENOMEM;
+
+                r = sd_bus_message_append(
+                        reply, "(iso)",
+                        link->ifindex,
+                        strempty(link->ifname),
```

Comment:
you can drop the strempty() btw, sd_bus_message_append() does that anyway for NULL strings